jerry-core/ecma/operations/ecma-promise-object.c

```diff
@@ -181,6 +181,23 @@ ecma_is_resolver_already_called (ecma_object_t *promise_obj_p) /**< promise */
   return (ecma_promise_get_flags (promise_obj_p) & ECMA_PROMISE_ALREADY_RESOLVED) != 0;
 } /* ecma_is_resolver_already_called */
 
+/**
+ * HostPromiseRejectionTracker
+ *
+ * See also: ES11 25.6.1.9
+ */
+static void
+ecma_promise_track_rejection (ecma_object_t *obj_p, /**< promise */
+                              ecma_value_t reason) /**< reason for reject */
+{
+  JERRY_ASSERT (ecma_is_promise (obj_p));
+
+  if (!(ecma_promise_get_flags (obj_p) & ECMA_PROMISE_HANDLED))
+  {
+    jerry_port_report_unhandled_promise_rejection (ecma_make_object_value (obj_p), reason);
```

Comment:
The name seems a bit counterintuitive to me. If the abstract operation of the spec is called HostPromiseRejectionTracker and the ecma core function is called `ecma_promise_track_rejection`, then why is the port function called `jerry_port_report_unhandled_promise_rejection`? Why report? Why unhandled?

Wouldn't be `jerry_port_track_promise_rejection` more natural?

Reply:
> Why report? Why unhandled?

  - A typical implementation of [HostPromiseRejectionTracker ](https://www.ecma-international.org/ecma-262/11.0/#sec-host-promise-rejection-tracker)might try to notify developers of unhandled rejections,
 
 I wasn't sure with the name, but I think `jerry_port_track_promise_rejection` is fine